### PR TITLE
Import, Export, Reset cmdlet added and vNet fixes 

### PR DESCRIPTION
**Pull request by SiddharthChatrolaMs**
- Azure Redis Cache
  - New cmdlet added for Reset-AzureRmRedisCache 
  - New cmdlet added for Export-AzureRmRedisCache 
  - New cmdlet added for Import-AzureRmRedisCache 
  - Modified cmdlet New-AzureRmRedisCache to include parameter change for vNet

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@SiddharthChatrolaMs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sichatro). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@SiddharthChatrolaMs Currently on hold for the netcore update

File: src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.RedisCache.Properties;
+    using Microsoft.Azure.Management.Redis.Models;
+    using RebootTypeStrings = Microsoft.Azure.Management.Redis.Models.RebootType;
+
+    [Cmdlet(VerbsCommon.Reset, "AzureRmRedisCache"), OutputType(typeof(bool))]
+    public class ResetAzureRedisCache : RedisCacheCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs]: SupportsShouldProcess=true

File: src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!Force.IsPresent)
```
Comment (markcowl) [inline src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs]: Do you really need a Force parameter here?  SImplu using ```if (ShouldProcess(...)```  should do what you need, unless there are special conditions the user would need to confirm at runtime.

**Comment by markcowl** [comment]
@SiddharthChatrolaMs A couple of comments, otherwise LGTM

File: src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs (hunk 1 above)
Comment (SiddharthChatrolaMs) [inline src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs]: "SupportsShouldProcess=true" what is usage of this parameter?

Comment (markcowl) [inline src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs]: This is a property of the Cmdlet attribute.  It indicates that the cmdlet supports SHouldProcess.  Any cmdlet that calls Cmdlet.ShouldProcess needs to have this attribute.

**Comment by markcowl** [comment]
@SiddharthChatrolaMs Ping.  Please respond to review comments - we are forcing correct usage of ShouldProcess for new cmdlets.

**Comment by markcowl** [comment]
@SiddharthChatrolaMs Please update the PR (there are merge conflicts) and respond to review comments.

File: src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs (hunk 2 above)
Comment (SiddharthChatrolaMs) [inline src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs]: Yes we do need Force here.

File: src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs (hunk 1 above)
Comment (SiddharthChatrolaMs) [inline src/ResourceManager/RedisCache/Commands.RedisCache/Commands/ResetAzureRedisCache.cs]: added SupportsShouldProcess = true

**Comment by SiddharthChatrolaMs** [comment]
Send https://github.com/Azure/azure-powershell/pull/2559 for targeted release branch

**Pull request closed without merging**